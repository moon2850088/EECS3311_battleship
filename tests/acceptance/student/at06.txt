--medium no win
debug_test(medium)
fire([A,1])
fire([A,2])
fire([A,3])
fire([B,1])
fire([B,2])
fire([B,3])
fire([B,4])
fire([B,5])
fire([B,6])
bomb([D,3],[D,2])
bomb([E,3],[E,2])
bomb([E,4],[E,2])
bomb([E,3],[E,2])
bomb([E,5],[E,6])
fire([C,1])
fire([C,2])
fire([C,3])
fire([C,4])
fire([C,5])
fire([C,6])
fire([D,1])
fire([D,2])
