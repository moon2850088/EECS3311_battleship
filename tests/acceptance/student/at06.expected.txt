  state 0 OK -> Start a new game
->debug_test(medium)
  state 1 OK -> Fire Away!
     1  2  3  4  5  6
  A  _  _  _  _  _  _
  B  _  _  _  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 0/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([A, 1])
  state 2 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  _  _  _  _  _
  B  _  _  _  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 1/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([A, 2])
  state 3 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  _  _  _  _
  B  _  _  _  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 2/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([A, 3])
  state 4 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  _  _  _  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 3/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([B, 1])
  state 5 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  _  _  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 4/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([B, 2])
  state 6 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  _  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 5/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([B, 3])
  state 7 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  _  _  _
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 6/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([B, 4])
  state 8 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  _  _
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 7/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([B, 5])
  state 9 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  _
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 8/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([B, 6])
  state 10 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  _  h  h  _  _  _
  D  _  v  _  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 9/16
  Bombs: 0/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->v;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->bomb([D, 3],[D, 2])
  state 11 OK -> Hit! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  _  h  h  _  _  _
  D  _  X  O  _  _  _
  E  _  v  _  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 9/16
  Bombs: 1/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->v;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->bomb([E, 3],[E, 2])
  state 12 OK -> Hit! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  _  h  h  _  _  _
  D  _  X  O  _  _  _
  E  _  X  O  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 9/16
  Bombs: 2/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->bomb([E, 4],[E, 2])
  state 13 Bomb coordinates must be adjacent -> Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  _  h  h  _  _  _
  D  _  X  O  _  _  _
  E  _  X  O  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 9/16
  Bombs: 2/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->bomb([E, 3],[E, 2])
  state 14 Already fired there -> Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  _  h  h  _  _  _
  D  _  X  O  _  _  _
  E  _  X  O  _  _  _
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 9/16
  Bombs: 2/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->bomb([E, 5],[E, 6])
  state 15 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  _  h  h  _  _  _
  D  _  X  O  _  _  _
  E  _  X  O  _  O  O
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 9/16
  Bombs: 3/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([C, 1])
  state 16 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  O  h  h  _  _  _
  D  _  X  O  _  _  _
  E  _  X  O  _  O  O
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 10/16
  Bombs: 3/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->v
    2x1: [C, 2]->h;[C, 3]->h
    1x1: [F, 4]->v
->fire([C, 2])
  state 17 OK -> Hit! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  O  X  h  _  _  _
  D  _  X  O  _  _  _
  E  _  X  O  _  O  O
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 11/16
  Bombs: 3/3
  Score: 0/6 (Total: 0/6)
  Ships: 0/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->h
    1x1: [F, 4]->v
->fire([C, 3])
  state 18 OK -> 2x1 ship sunk! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  O  X  X  _  _  _
  D  _  X  O  _  _  _
  E  _  X  O  _  O  O
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 12/16
  Bombs: 3/3
  Score: 2/6 (Total: 2/6)
  Ships: 1/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->X
    1x1: [F, 4]->v
->fire([C, 4])
  state 19 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  O  X  X  O  _  _
  D  _  X  O  _  _  _
  E  _  X  O  _  O  O
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 13/16
  Bombs: 3/3
  Score: 2/6 (Total: 2/6)
  Ships: 1/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->X
    1x1: [F, 4]->v
->fire([C, 5])
  state 20 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  O  X  X  O  O  _
  D  _  X  O  _  _  _
  E  _  X  O  _  O  O
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 14/16
  Bombs: 3/3
  Score: 2/6 (Total: 2/6)
  Ships: 1/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->X
    1x1: [F, 4]->v
->fire([C, 6])
  state 21 OK -> Miss! Keep Firing!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  O  X  X  O  O  O
  D  _  X  O  _  _  _
  E  _  X  O  _  O  O
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 15/16
  Bombs: 3/3
  Score: 2/6 (Total: 2/6)
  Ships: 1/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->X
    1x1: [F, 4]->v
->fire([D, 1])
  state 22 OK -> Miss! Game Over!
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  O  X  X  O  O  O
  D  O  X  O  _  _  _
  E  _  X  O  _  O  O
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 16/16
  Bombs: 3/3
  Score: 2/6 (Total: 2/6)
  Ships: 1/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->X
    1x1: [F, 4]->v
->fire([D, 2])
  state 23 Game not started -> Start a new game
     1  2  3  4  5  6
  A  O  O  O  _  _  _
  B  O  O  O  O  O  O
  C  O  X  X  O  O  O
  D  O  X  O  _  _  _
  E  _  X  O  _  O  O
  F  _  v  _  v  _  _
  Current Game (debug): 1
  Shots: 16/16
  Bombs: 3/3
  Score: 2/6 (Total: 2/6)
  Ships: 1/3
    3x1: [D, 2]->X;[E, 2]->X;[F, 2]->v
    2x1: [C, 2]->X;[C, 3]->X
    1x1: [F, 4]->v
